src/id.rs

```diff
@@ -284,10 +287,45 @@ impl PartialOrd for P2pNode {
 
 impl Ord for P2pNode {
     fn cmp(&self, other: &Self) -> Ordering {
-        self.public_id.cmp(&other.public_id)
+        let self_socket_addr = SocketAddrWrapper(self.connection_info.peer_addr);
```

Comment:
Not ideal, you still have a discrepency between Ord and Eq.
Either you implement Ord and Eq or you derive Ord and Eq.

If you want to go for the wrapper implementation, probably better to store that wrapper as oppose to constructing it on each comparison, also your wrapper may be better as ConnectionInfoWrapper since that is what you store.
Then you can Derive Ord,Eq,Hash on P2pNode and implement them on ConnectionInfoWrapper.